File: 6_multiplevel_inheritance.py

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age
        
    def display(self):
        print(self.name)
        print(self.age)
        

class Student(Person):
    def __init__(self, name, age, rollno):
        super().__init__(name, age)
        self.rollno = rollno
        
    def display(self):
        Person.display(self)
        print(self.rollno)
        
    
class UniversityStudent(Student):
    def __init__(self, name, age, rollno, major):
        super().__init__(name,age,rollno)
        self.major = major
        
    def display(self):
        Student.display(self)
        print(self.major)
        

obj = UniversityStudent("Ishtiaq", 20, 101, "CSE")

obj.display()

File: 6_multiplevel_inheritance.cpp
#include<bits/stdc++.h>
using namespace std;

class Person{
    public:
        string name;
        int age;
        Person(string name, int age){
            this->name = name;
            this->age = age;
        }
};

class Student: public Person{
    public:
    int rollno;
    Student(string name,int age, int rollno):Person(name,age){
        this->rollno = rollno;
    }

};
class UniversityStudent: public Student{
   public:
       string major;

       UniversityStudent(string name,int age,int rollno,string major):
       Student(name,age,rollno){
        this->major = major;
       }

    void display(){
        cout<<"Name: "<<name<<endl;
        cout<<"Age: "<<age<<endl;
        cout<<"Rollno: "<<rollno<<endl;
        cout<<"Major: "<<major<<endl;
    }
};
int main(){
    UniversityStudent s1("Ishtiaq",23,12,"CSE");
    s1.display();

}